pkgs/by-name/te/terminusdb/package.nix

```diff
@@ -0,0 +1,149 @@
+    repo = "jwt_io";
+    rev = "v${finalAttrs.jwtVersion}";
+    hash = "sha256-J1FlKVGDVrnBXXvPK9VIcxHwlFDcZyQ1iorS4ornIOc=";
+    postFetch = ''
+      # Remove the `jwks_endpoint` test that requires internet access
+      sed -i '84,89d' $out/tests/jwt_io_jwt.plt
+    '';
+  };
+
+  dashboard = fetchzip {
+    url = "https://github.com/terminusdb/terminusdb-dashboard/releases/download/v${finalAttrs.dashboardVersion}/release.tar.gz";
+    hash = "sha256-x0iOAUFWSJXnVgmkRkM6tiQZVYdQD8NVxGlzW/LmgpE=";
+    stripRoot = false;
+  };
+
+  swiPrologWithDeps = (swiProlog.overrideAttrs (finalSwiPrologAttrs: previousAttrs: {
+    pname = "terminusdb-swi-prolog";
+    buildInputs = previousAttrs.buildInputs ++ [libjwt];
+  })).override {
+    extraPacks = map (dep-path: "'file://${dep-path}'") [
+      finalAttrs.tus
+      finalAttrs.jwt_io
+    ];
+  };
+
+  src = fetchFromGitHub {
+    owner = "terminusdb";
+    repo = "terminusdb";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-5Wa0Z453+y3wbLNWKbd/9n0Dczgv4uXZGbgd/LHKe90=";
+    leaveDotGit = true;
+    postFetch = ''
+      # Will be used for `TERMINUSDB_GIT_HASH`
+      git -C $out rev-parse HEAD > $out/COMMIT
+      find "$out" -name .git -type d -exec rm -rf {} +
+    '';
+  };
+
+  cargoRoot = "src/rust";
+
+  cargoDeps = rustPlatform.importCargoLock {
+    # Copied from https://github.com/terminusdb/terminusdb/blob/v11.1.12/src/rust/Cargo.lock
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "juniper-0.15.11" = "sha256-kAzXJW5EdwpQds5uLWOjDhLganFquk8rbFjmqPvpj9w=";
+      "terminusdb-grpc-labelstore-client-0.1.0" = "sha256-OfxSnvWpFWwd1N2o9FwXVQ0VMBEqKa7mjtFoJSmPuFk=";
+    };
+  };
+
+  nativeBuildInputs = [
+    (with rustPlatform; [
+      cargoSetupHook
+      bindgenHook
+    ])
+    cargo
+    gmp
+    protobuf
+    libclang
+  ];
+
+  buildInputs = [
+    finalAttrs.swiPrologWithDeps
+  ];
+
+  checkTarget = "test";
+  doCheck = true;
+
+  # Remove tests that require networking
+  postPatch = ''
+    endpointTests="db clone fetch rebase pack optimize"
+    for test in $endpointTests; do
+      test+=_endpoint
+      echo "Removing tests for $test..."
+      sed -i "/:- begin_tests($test)./,/:- end_tests($test)./d" src/server/routes.pl
+    done
+  '';
+
+  preBuild = ''
+    export TERMINUSDB_GIT_HASH=$(cat $src/COMMIT)
+    export TERMINUSDB_DASHBOARD_PATH=$out/dashboard
+    export TERMINUSDB_JWT_ENABLED=true
+  '';
+
+  dontStrip = true;
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 terminusdb -t $out/bin
+    cp ${finalAttrs.dashboard} -r $out/dashboard
+    runHook postInstall
+  '';
+
+  passthru.tests = {
+    upstream-integration = callPackage ./tests.nix { };
+  };
+
+  meta = {
+    # Broken on Darwin
+    # aarch64 https://logs.ofborg.org/?key=nixos/nixpkgs.303209&attempt_id=383c1b9e-cdff-4d34-9375-4a242d566353
+    # x86_64 https://logs.ofborg.org/?key=nixos/nixpkgs.303209&attempt_id=ab659636-0d8a-43dc-82c3-2414de44b6bc
```

Comment:
I don't think those links are mean for long term storage. Please include the error instead. Maybe also upstream doesn't even support darwin?